lib/logflare/backends/adaptor/webhook_adaptor.ex

```diff
@@ -80,14 +81,27 @@ defmodule Logflare.Backends.Adaptor.WebhookAdaptor do
     @moduledoc false
     use Tesla, docs: false
 
-    plug(Tesla.Middleware.Telemetry)
-    plug(Tesla.Middleware.JSON)
-
     def send(opts) do
-      opts
-      |> Keyword.put_new(:method, :post)
-      |> Keyword.update(:headers, [], &Map.to_list/1)
-      |> request()
+      adaptor =
+        if Keyword.get(opts, :http) == "http1" do
+          {Tesla.Adapter.Finch, name: Logflare.FinchDefaultHttp1, receive_timeout: 5_000}
+        else
+          {Tesla.Adapter.Finch, name: Logflare.FinchDefault, receive_timeout: 5_000}
+        end
```

Comment:
`Finch` do not take care of it automatically? Because that is weird that downstream is needs to know which protocol it should use.

Reply:
Finch can handle both but without multiplexing
https://hexdocs.pm/finch/Finch.html#start_link/1-pool-configuration-options

Currently all our pools are http2 to leverage the multiplexing, so we'd need to have a separate http1 pool for that, separate from the http2 pools.
We could adjust the default pool to be both http1 and http2, but then we won't be able to get any multiplexing if the backend server supports it